ui/src/shared/utils/downloadTimeseriesCSV.ts

```diff
@@ -0,0 +1,100 @@
+import moment from 'moment'
+import {unparse} from 'papaparse'
+
+import {proxy} from 'src/utils/queryUrlGenerator'
+import {duration} from 'src/shared/apis/query'
+import replaceTemplates, {replaceInterval} from 'src/tempVars/utils/replace'
+import {timeSeriesToTableGraph} from 'src/utils/timeSeriesTransformers'
+
+import {Query, Template} from 'src/types'
+import {TimeSeriesServerResponse} from 'src/types/series'
+
+type CSV = Array<Array<number | string>>
+
+const downloadTimeseriesCSV = async (
+  queries: Query[],
+  templates: Template[]
+): Promise<void> => {
+  const responses = await Promise.all(
+    queries.map(query => fetchTimeseries(query, templates))
+  )
+
+  const csv = await timeseriesToCSV(responses)
+
+  downloadCSV(csv, csvName(csv))
+}
+
+const fetchTimeseries = async (
+  query: Query,
+  templates: Template[]
+): Promise<TimeSeriesServerResponse> => {
+  const source = query.queryConfig.source
+
+  let queryString = replaceTemplates(query.text, templates)
+
+  if (queryString.includes(':interval:')) {
+    const queryDuration = await duration(query.text, source)
+
+    queryString = replaceInterval(query.text, null, queryDuration)
+  }
+
+  const resp = await proxy({source: source.links.proxy, query: queryString})
+
+  return {response: resp.data}
+}
+
+const timeseriesToCSV = async (
+  responses: TimeSeriesServerResponse[]
+): Promise<CSV> => {
+  const tableResponse = await timeSeriesToTableGraph(responses)
+  const table = tableResponse.data
+
+  if (!table.length) {
+    throw new Error('no results')
+  }
+
+  const header = table[0]
+  const timeIndex = header.indexOf('time')
+
+  if (timeIndex < 0) {
+    throw new Error('could not find time index')
+  }
+
+  for (let i = 1; i < table.length; i++) {
+    // Convert times to a (somewhat) human readable ISO8601 string
+    table[i][timeIndex] = new Date(table[i][timeIndex]).toISOString()
+  }
+
+  return table
+}
+
+const csvName = (csv: CSV) => {
+  // Strange time format since `:` can't be used in a filename
+  const now = moment().format('YYYY-MM-DD-HH-mm')
+
+  let columns = []
+
+  if (csv.length) {
+    columns = csv[0]
+      .filter(col => col !== 'time')
+      .map((col: string) => col.replace('.', '_'))
+  }
+
+  return `${now} ${columns.join(' ')}.csv`
```

Comment:
hm good point.

In the data explorer, it used the database, retention policy, and measurement to name the exported CSV. Since this CSV can contain multiple sources, it no longer made sense to use that naming scheme.

I opted for this instead, but am open to suggestions. What about just the date and some generic message? Like `2018-08-28 Chronograf Data.csv` or something

Reply:
that looks good to me. Maybe include the timestamp incase the download more than one in a day?

Reply:
sounds good